### PR TITLE
add minSponsorTokens to limit max redeem tokens

### LINKED ISSUE #17: Redeem allows you to send a txn that would take you under the min sponsor size
n/t

### PR DESCRIPTION
**Pull request by adrianmcli**
Fixes #17 

This limits the ability to redeem tokens such that your resulting balance will be less than the `minSponsorTokens` count.

### CODE REVIEW

File: features/manage-position/Redeem.tsx [hunk 1]
```diff
@@ -24,7 +29,8 @@ const Redeem = () => {
   const [error, setError] = useState<Error | null>(null);
 
   const tokensToRedeemFloat = isNaN(parseFloat(tokensToRedeem)) ? 0 : parseFloat(tokensToRedeem);
-  const maxRedeem = Math.min((syntheticBalance || 0), (borrowedTokens || 0));
+  const tokensAboveMin = borrowedTokens && empState.minSponsorTokens ? borrowedTokens - weiToNum(empState.minSponsorTokens) : 0;
+  const maxRedeem = Math.min((syntheticBalance || 0), (borrowedTokens || 0), (tokensAboveMin ||0));
```
Comment (mrice32) [inline features/manage-position/Redeem.tsx:33]: Does this prevent them from redeeming _all_ of their tokens?

Comment (adrianmcli) [reply features/manage-position/Redeem.tsx:33]: Yes, I believe so. Should we make an exception for that? Basically the rule should only apply if they are not unwinding their entire position?

Comment (adrianmcli) [reply features/manage-position/Redeem.tsx:33]: Currently, say `minSponsorTokens` is 1000, and they hold 1200 tokens, then:

- `syntheticBalance` = 1200
- `borrowedTokens` = 1200
- `tokensAboveMin` = 200

They would only be able to redeem 200.

Comment (mrice32) [reply features/manage-position/Redeem.tsx:33]: I think we should make an exception for that. They should be able to completely unwind their position IMO.

Comment (mrice32) [reply features/manage-position/Redeem.tsx:33]: So in this case we would want to basically say, you can redeem between 0 and 200 tokens or all 1200 tokens. This makes the logic pretty annoying, though :/

Comment (adrianmcli) [reply features/manage-position/Redeem.tsx:33]: lol yes this is very annoying.

**Comment by adrianmcli** [comment]
@mrice32 that should do it, it should allow complete unwinding as an "alternative" path.

And yikes, if we're going to support this, we really need to use a proper form validation framework ASAP. 🤮🤮🤮

**Pull request merged**